File: runme/mcgen/local/datan/exasrc/e3gam.c
#include <stdio.h>
#include "datsrc.h"

main(void)
{
    static integer iswit;
    static double a, b, f, w, x, z;
/* identify program to user */
    printf("%s\n", "Program E3GAM demonstrates use of");
    printf("%s\n", "GBETAF,GINCGM and GINCBT");
    printf("%s\n", "and provides interactively numerical values");
    printf("%s\n", "of the beta function, the incomplete gamma");
    printf("%s\n\n", "function, and the incomplete beta function.");
    printf("%s\n", "Choose:");
    printf("%s\n", "1 - beta function");
    printf("%s\n", "2 - incomplete gamma function");
    printf("%s\n", "3 - incomplete beta function");
    printf("%s", "> ");
    scanf("%li", &iswit);
    if (iswit == 1) {
/* beta */
        printf("%s\n", "Enter z (> 0.):");
        printf("%s", "> ");
        scanf("%lf", &z);
        printf("%s\n", "Enter w (> 0.):");
        printf("%s", "> ");
        scanf("%lf", &w);
        f = gbetaf_(&z, &w);
        printf("%s%lf\n", "beta(z,w) = ", f);
    } else if (iswit == 2) {
/* incomplete gamma */
        printf("%s\n", "Enter a (>0.):");
        printf("%s", "> ");
        scanf("%lf", &a);
        printf("%s\n", "Enter x (>0.):");
        printf("%s", "> ");
        scanf("%lf", &x);
        f = gincgm_(&a, &x);
        printf("%s%lf\n", "P(a,x) = ", f);
    } else if (iswit == 3) {
/* incomplete beta */
        printf("%s\n", "Enter a, (0.<a<1.):");
        printf("%s", "> ");
        scanf("%lf", &a);
        printf("%s\n", "Enter b, (0.<b<1.):");
        printf("%s", "> ");
        scanf("%lf", &b);
        printf("%s\n", "Enter x, (0.<x<1.):");
        printf("%s", "> ");
        scanf("%lf", &x);
        f = gincbt_(&a, &b, &x);
        printf("%s%lf\n", "I_x (a,b) = ", f);
    }
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e3gam.f
      PROGRAM E3GAM
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
C identify program to user
      WRITE(*,*)' Program E3GAM demonstrates use of'
      WRITE(*,*)' GBETAF,GINCGM and GINCBT'
      WRITE(*,*)' and provides interactively numerical values'
      WRITE(*,*)' of the beta function, the incomplete gamma'
      WRITE(*,*)' function, and the incomplete beta function'
      WRITE(*,*)' '
      WRITE(*,*)' Choose'
      WRITE(*,*)' 1 - beta function'
      WRITE(*,*)' 2 - incomplete gamma function'
      WRITE(*,*)' 3 - incomplete beta function'
      WRITE(*,*)' >'
      READ(*,*)ISWIT
      IF(ISWIT.EQ.1) THEN
C beta
        WRITE(*,*)' Enter z (>= 0.)'
        WRITE(*,*)' >'
        READ(*,*)Z
        WRITE(*,*)' Enter w (>= 0.)'
        WRITE(*,*)' >'
        READ(*,*)W
        F=GBETAF(Z,W)
        WRITE(*,*)' beta(z,w) = ',F
      ELSE IF(ISWIT.EQ.2) THEN
C incomplete gamma
        WRITE(*,*)' Enter a (>0.)'
        WRITE(*,*)' >'
        READ(*,*)A
        WRITE(*,*)' Enter x (>0.)'
        WRITE(*,*)' >'
        READ(*,*)X
        F=GINCGM(A,X)
        WRITE(*,*)' P(a,x) = ',F
      ELSE IF(ISWIT.EQ.3) THEN
C incomplete beta
        WRITE(*,*)' Enter a, (0.<a<1.) '
        WRITE(*,*)' >'
        READ(*,*)A
        WRITE(*,*)' Enter b, (0.<b<1.)'
        WRITE(*,*)' >'
        READ(*,*)B
        WRITE(*,*)' Enter x, (0.<x<1.)'
        WRITE(*,*)' >'
        READ(*,*)X
        F=GINCBT(A,B,X)
        WRITE(*,*)' I_x (a,b) = ',F
      END IF
      END
